src/libraries/Common/src/Interop/Unix/System.Native/Interop.SchedGetCpu.cs

```diff
@@ -9,6 +9,7 @@ internal static partial class Interop
     internal unsafe partial class Sys
     {
         [LibraryImport(Interop.Libraries.SystemNative, EntryPoint = "SystemNative_SchedGetCpu")]
+        [SuppressGCTransition]
         internal static partial int SchedGetCpu();
```

Comment:
Blocking, or even taking too long, diminish the usefulness of this API as it would increase the chance of a context switch.
I think we can assume that most implementations will try to be fast. 

Long `SuppressGCTransition` calls could be a concern as they could cause GC suspension pauses, especially if called in a loop. But this will not take long. On modern hardware this is basically a read form a special register and perhaps some unpacking/reinterpreting the value as an integer.

Also, in degenerate/slow cases we will be accordingly more aggressive with using thread-local cache, thus "call in a loop" scenario will be mostly fetching the cached value.